Check for required PHP extensions

xh3n1 opened the pull request:
<!---Thanks for contributing to phpList!-->

## Related Issue
<!--- If it fixes an open issue on Mantis , please include a link to the issue here. -->
https://mantis.phplist.org/view.php?id=19837

samtuke commented on public_html/lists/admin/index.php:
```diff
@@ -23,6 +23,42 @@
+    }
+}
+if (count($notinstalled) > 0) {
+    $message = "The following PHP extensions are missing:" . '<br>';
```
Should this be translatable?

xh3n1 replied:
It cannot be translated as it stands above required language files.

samtuke commented on public_html/lists/admin/index.php:
```diff
@@ -23,6 +23,42 @@
 require_once dirname(__FILE__).'/inc/random_compat/random.php';
 
+// Check if required extensions are installed.
+$phpExtensions = array(
```
How about naming this $requiredExtensions for clarity?

xh3n1 replied:
@samtuke OK

bramley commented:
@michield @samtuke Despite what the documentation might say, some of the php extensions are not mandatory because the code sometimes checks before trying to use it.

Examples are curl and mbstring, see index.php
```
if (!function_exists('mb_strtolower')) {
    function mb_strtolower($string)
    {
        return strtolower($string);
    }
}

```

When this change is included in a release, if someone who does not have all of these extensions upgrades then phplist is going to stop working for them. I think that any change to making extensions mandatory needs a bit more thought.

xh3n1 commented:
@bramley 
There are not always checks for functions before trying to use them, which leads to blank pages etc.
Following your comment here, I created this PR https://github.com/phpList/phplist3/pull/513
where I moved the code to checkprerequisites.php, and removed some extensions that I think are more optional( those that have if-else step around functions).
